Cython/Build/Dependencies.py

```diff
@@ -1040,6 +1044,23 @@ def copy_to_build_dir(filepath, root=os.getcwd()):
                     dir = os.path.dirname(c_file)
                     safe_makedirs_once(dir)
 
+                # write out the depfile, if requested
+                if depfile:
+                    dependencies = deps.all_dependencies(source)
+                    src_base_dir, _ = os.path.split(source)
+                    # paths below the base_dir are relative, otherwise absolute
+                    paths = []
+                    for fname in dependencies:
+                        if fname.startswith(src_base_dir):
```

Comment:
Still not quite enough. If you have paths `/a/b/c` and `/a/b/cd`, then the second starts with the first but is not an actual parent directory. There's [`os.path.commonpath()`](https://docs.python.org/3/library/os.path.html#os.path.commonpath), but that was only added in Py3.5.  For Py2.7, we'd then need a fallback. However, you could simply append `os.sep` to `src_base_dir` to make the comparison safer.
```suggestion
                    src_base_dir, _ = os.path.split(source)
                    if not src_base_dir.endswith(os.sep):
                        src_base_dir += os.sep
                    # paths below the base_dir are relative, otherwise absolute
                    paths = []
                    for fname in dependencies:
                        if fname.startswith(src_base_dir):
```